### PR TITLE
draft for mgmt README

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
Fix https://github.com/Azure/azure-libraries-for-java/issues/1131

First draft for README

Let me know if there is more worth putting to key concepts and examples.

### CODE REVIEW

File: sdk/management/README.md [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+By default, Azure Active Directory token authentication depends on correct configure of following environment variables.
+
+- `AZURE_CLIENT_ID` for Azure client ID.
+- `AZURE_TENANT_ID` for Azure tenant ID.
+- `AZURE_CLIENT_SECRET` or `AZURE_CLIENT_CERTIFICATE_PATH` for client secret or client certificate.
+
+In addition, Azure subscription ID can be configured via environment variable `AZURE_SUBSCRIPTION_ID`.
+
+See [Authenticate][authenticate] for more options.
```
Comment (weidongxu-microsoft) [inline sdk/management/README.md:180]: Point to AUTH.md later.

File: sdk/management/azure/pom.xml [hunk 2]
```diff
@@ -12,7 +12,7 @@
     <relativePath>../pom.xml</relativePath>
   </parent>
 
-  <artifactId>azure</artifactId>
+  <artifactId>azure-management</artifactId>
```
[the remark this answers is not in the document]
Comment (ChenTanyi) [reply sdk/management/azure/pom.xml:15]: Does the azure-parent pom also need to be changed?

[the remark this answers is not in the document]
Comment (weidongxu-microsoft) [reply sdk/management/azure/pom.xml:15]: Yes. I think so.

File: sdk/management/README.md [hunk 3]
```diff
@@ -0,0 +1,221 @@
+# Azure management client library for Java
+
+The Azure Management Library for Java is a higher-level, object-oriented API for *managing* Azure resources,
```
Comment (xccc-msft) [inline sdk/management/README.md:3]: Which should be the expected for us in this repo, `Azure management client library for Java`, `Azure Management Library for Java`? Or using `libraries` as we did in previous repo? To distinguish the dataplane SDKs and auto-generated Lite SDKs.

Comment (weidongxu-microsoft) [reply sdk/management/README.md:3]: The title is due to this https://github.com/Azure/azure-sdk-for-java/blob/master/eng/.docsettings.yml#L16
Currently not want to touch that yet.

Given this README is located for all mgmt sdks, yes, I agree `libraries` would be better.

File: sdk/management/README.md [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+
+### Include optional packages
+
+[Azure Identity][azure_identity] provides Azure Active Directory token authentication support across the Azure SDK.
+
+[//]: # ({x-version-update-start;com.azure:azure-identity;current})
+```xml
+<dependency>
+  <groupId>com.azure</groupId>
+  <artifactId>azure-identity</artifactId>
```
Comment (yungezz) [inline sdk/management/README.md:36]: is azure-identity optional package? I can get the point our mgmt libs not depends on it. but from user's perspective, how to make it clear to them that they need azure-identity or TokenCredential before performing any mgmt operation. Maybe link to auth.md somewhere?

File: sdk/management/README.md [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+  <version>1.1.0-beta.3</version>
+</dependency>
+```
+[//]: # ({x-version-update-end})
+
+[Azure Core Netty HTTP client][azure_core_http_netty] is a plugin for [Azure Core][azure_core] HTTP client API.
```
Comment (yungezz) [inline sdk/management/README.md:42]: i think it's worth to a separate section to let user know they need choose from 2 http client by themselves, and reference to each of them

File: sdk/management/README.md [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+or checkout [StackOverflow for Azure Java SDK](http://stackoverflow.com/questions/tagged/azure-java-sdk).
+
+### HTTP client
+
+An `HttpClient` implementation must exist on the classpath.
+See [Include optional packages](#include-optional-packages).
+
+### Authentication
+
+By default, Azure Active Directory token authentication depends on correct configure of following environment variables.
```
Comment (yungezz) [inline sdk/management/README.md:172]: how about move this earlier, my point is that as a user, when i read or follow major steps in the readme, I can have a rough end to end idea of complete a mgmt task, from auth to config, to service features, then logging etc advanced features.

**Comment by yungezz** [comment]
is it necessary to have a list of java mgmt sdks, then reference to it from here for users' easy search?

File: sdk/management/README.md (hunk 4 above)
Comment (weidongxu-microsoft) [reply sdk/management/README.md:36]: It is optional. User can do their own OAUTH, without any lib. I think this is the exact reason that requires us not to include azure-identity module.

File: sdk/management/README.md (hunk 5 above)
Comment (weidongxu-microsoft) [reply sdk/management/README.md:42]: We can optionally refer to https://github.com/Azure/azure-sdk-for-java/wiki/HTTP-clients

I've checked other libs when writing this, they generally only mention it https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/storage/azure-storage-common#default-http-client

File: sdk/management/README.md (hunk 6 above)
Comment (weidongxu-microsoft) [reply sdk/management/README.md:172]: Currently following other client README structure dictated by https://github.com/Azure/azure-sdk-for-java/blob/master/eng/.docsettings.yml#L16

When AUTH.md is ready, we can refer it in first line.

Comment (weidongxu-microsoft) [reply sdk/management/README.md]: I've moved this section to "Getting started" at top. Also added sample code.

File: sdk/management/azure/pom.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (weidongxu-microsoft) [reply sdk/management/azure/pom.xml:15]: Also change parent to "azure-management-parent".

**Pull request merged**